### DISCUSSION TITLE
Connect backend to fill data catalogue

### DISCUSSION THREAD

**Discussion by azrimadit** [TerriaJS/terriajs#5898] [Q&A] [answered]
Hello, i'm alex.
I want to ask something. I've created a backend server using express. I store the data to the backend and make an API to call the res.json that contains json like simple.json. Where I can input the API in TerriaMap configuration?

Thanks for the answer.

Alex

**Comment by zoran995** [comment, marked as answer]
Hi @alex192345 
It should work if you add a link to `initializationUrls` in config file, like it is done for nationalmap
https://github.com/TerriaJS/nationalmap/blob/cff947ae3d65edbc51ef3ff4ac9cb2c7b91e6ad2/wwwroot/config.json#L2-L4

**Reply by azrimadit** [reply]
Hello @zoran995 !
Thanks for the reply. I've made a backend and the res.json output like simple.json so the explore data won't read the json file again, but the axios to the backend and catch the json API. Can you help me to solve this?

Thank you

Alex

**Reply by zoran995** [reply]
In that case, you will need to update the way terria handles initialization URLs. These two parts are most relevant here we go through the list of initializationURLs and call function `generateInitializationUrl`
https://github.com/TerriaJS/terriajs/blob/c83e90c4aab4460e5a3943925b4ff750448b078c/lib/Models/Terria.ts#L727-L737

and then in `generateInitializationUrl` we check if it doesn't end with `.json`, if it doesn't end with `.json` it's considered as the name of the file (probably the fix would be to check if it's an URL)
https://github.com/TerriaJS/terriajs/blob/c83e90c4aab4460e5a3943925b4ff750448b078c/lib/Models/Terria.ts#L1730-L1749

**Reply by azrimadit** [reply]
Hi @zoran995 thanks for the help and answers. I have edited jsonValue with:
`jsonValue = await (await axios.get(initSource.initURL)).data`
and the initSource.initURL will return the backend API.

Thanks a lot. It help me so much.

Sincerely,

Alex
